### PR TITLE
PBjs Core Targeting: allow non-string (eg. numeric) targeting segments

### PR DESCRIPTION
**Pull request by jimdigriz**
[Documentation shows a numeric example](https://docs.prebid.org/dev-docs/add-rtd-submodule.html#gettargetingdata) which causes an exception as we try to call `.split(',')`.

## Type of change
- [x] Bugfix

## Description of change
Allows non-string values to be passed.

### CODE REVIEW

**Comment by jimdigriz** [comment]
Can I do anything to help the review process? This blocks me on pushing forward on https://github.com/prebid/Prebid.js/pull/6309 and https://github.com/prebid/Prebid.js/pull/6310.

**Comment by Rothalack** [comment]
Hey, sorry for the delay, I had a busy week. I'm testing this now.

This will need a second review as well because it edits a core file.

**Comment by Rothalack** [comment]
This appears to be working fine, but I'm not certain the way I tested would really demonstrate. @jimdigriz  Do you have any set targeting lines you used to test? I could compare and be more certain.

@ChrisHuie Since this is a core file, we'll want a second review for this right? What do you think.

**Comment by jimdigriz** [comment]
Can someone show me how to run *just* the targeting unit test on its own? The following fails with a lot of errors (I think something about `fun-hook` needs to be initialised beforehand):
```
gulp test-coverage --file test/spec/unit/core/targeting_spec.js && gulp view-coverage
```

It was really helpful (and a *lot* faster) being able to do this for our analytic/video/rtd modules development, and I would really like to do this for targeting if I am expected to provide further coverage/unit tests.

**Comment by jimdigriz** [comment]
@ChrisHuie or @Rothalack, can either of you [help and teach me how to run just the targeting unit/coverage tests](https://github.com/prebid/Prebid.js/pull/7160#issuecomment-881274033)?

**Comment by Rothalack** [comment]
> @ChrisHuie or @Rothalack, can either of you [help and teach me how to run just the targeting unit/coverage tests](https://github.com/prebid/Prebid.js/pull/7160#issuecomment-881274033)?

I also attempted to run this and I think I got to the same place as you. I am also unsure.

**Comment by patmmccann** [comment]
I think you change if to if.only locally

**Comment by jimdigriz** [comment]
> I think you change if to if.only locally to run a single test; we saw an example of that accidentally get merged a couple of months back. You can also just delete most of the spec files for your local tests. Or did you mean running only a subset of the tests on pr submission here? That isn't something that is going to fly.

I am looking for an 'official' way to run the unit tests that cover the code I am expected to work on. Hacking and editing unrelated files that are not going to be part of the PR is a somewhat unconventional solution. Asking contributors to guess (I have no idea what to do, I am a Perl/Erlang guy forced to do JavaScript) what to edit ("'fun-hook' not initialised" is not exactly an actionable errror) is something similarly that is not going to fly.

For example for an analytics/rtd/adservervideo module all I needed during development was to run verbatim each time:

    gulp test-coverage --file 'test/spec/modules/adloox{AnalyticsAdapter,AdServerVideo,RtdProvider}_spec.js' && gulp view-coverage

Once I had this working (takes less than 20 seconds to run each time) I am more than happy to burn time running the full suite for the purposes of PR submission.

No need to edit anything, it was officially supported (though undocumented) according to your `gulpfile.js`.

I am looking for what I need to type to run something that it is *just* the targeting spec, I do not want to test the user cookie/privacy modules or spotx bidder:

    gulp test-coverage --file test/spec/unit/core/targeting_spec.js && gulp view-coverage

It is unfair to ask contributors to run a five minutes a cycle to run on my laptop every time to check the impact of a single keypress.

I am not proposing deleting or disabling unit tests for the PR. I am though asking for help from those with experience on how to get the project unit tests working an a manner that supports development and not just make it more difficult or less enjoyable.

Thanks

**Comment by jimdigriz** [comment]
@patmmccann @Rothalack do you know of anyone in the project who can provide guidance how to run *just* the `targeting_spec.js` unit test for the purposes of development?

**Comment by Rothalack** [comment]
I wish I had a good answer for this @jimdigriz , I'm a little out of my depth on this.

**Comment by patmmccann** [comment]
If you are talking about running limited tests locally, just delete all the spec files you don't want. If you are talking about submitting to the project, the "user cookie/privacy modules or spotx bidder" tests remain important, as they may depend on the files you are editing in ways that are difficult to predict.

**Comment by jimdigriz** [comment]
> If you are talking about running limited tests locally, just delete all the spec files you don't want.

@patmmccann I have done as you instructed, it does not work. What do I do now?

```
alex@aineko:~/src/Prebid.js$ git log -n2 HEAD
commit 3eadb8f8983104df75986cb1591b00120422a66f (HEAD -> fix/targeting-non-string, origin/fix/targeting-non-string, coremem/fix/targeting-non-string)
Author: Alexander Clouter <alex@coremem.com>
Date:   Sat Jun 19 14:08:09 2021 +0100

    targeting: allow non-string (eg. numeric) targeting segments
    
    Documentation[1] shows a numeric example which causes an exception as we
    try to call .split(',').
    
    [1] https://docs.prebid.org/dev-docs/add-rtd-submodule.html#gettargetingdata

commit 541fc70ada90703d628a0e520899900a6cecc73e (tag: 5.7.0)
Author: Matt Kendall <1870166+mkendall07@users.noreply.github.com>
Date:   Thu Jul 29 14:32:07 2021 -0400

    5.7.0 release

alex@aineko:~/src/Prebid.js$ find test/spec -name '*_spec.js' \! -name 'targeting_spec.js' -delete

alex@aineko:~/src/Prebid.js$ gulp test-coverage && gulp view-coverage
[15:55:14] Using gulpfile ~/src/adloox/Prebid.js/gulpfile.js
[15:55:14] Starting 'test-coverage'...
[15:55:14] Starting 'clean'...
[15:55:14] Finished 'clean' after 33 ms
[15:55:14] Starting 'testCoverage'...
31 07 2021 15:55:14.421:WARN [karma-server]: Passing raw CLI options to `new Server(config, done)` is deprecated. Use `parseConfig(configFilePath, cliOptions, {promiseConfig: true, throwErrors: true})` to prepare a processed `Config` instance and pass that as the `config` argument instead.

START:
⚠ ｢wdm｣: 
ℹ ｢wdm｣: Compiled with warnings.
ℹ ｢wdm｣: Compiling...
⚠ ｢wdm｣: 
ℹ ｢wdm｣: Compiled with warnings.
31 07 2021 15:55:20.683:INFO [karma-server]: Karma v6.3.4 server started at http://localhost:9876/
31 07 2021 15:55:20.683:INFO [launcher]: Launching browsers ChromeHeadless with concurrency 5
31 07 2021 15:55:20.690:INFO [launcher]: Starting browser ChromeHeadless
31 07 2021 15:55:21.122:INFO [Chrome Headless 92.0.4515.107 (Linux x86_64)]: Connected on socket SYlwRPV7ywOcSeDpAAAB with id 93973362
31 07 2021 15:55:21.488:WARN [reporter]: SourceMap position not found for trace: Uncaught TypeError: window.pbjs.processQueue is not a function
at http://localhost:9876/base/test/test_index.js?50c2d46de641c201d93f959a15507e2c93686a45:74239:13

TypeError: window.pbjs.processQueue is not a function
    at Object.<anonymous> (http://localhost:9876/base/test/test_index.js?50c2d46de641c201d93f959a15507e2c93686a45:74239:13)
    at __webpack_require__ (http://localhost:9876/base/test/test_index.js?50c2d46de641c201d93f959a15507e2c93686a45:20:30)
    at http://localhost:9876/base/test/test_index.js?50c2d46de641c201d93f959a15507e2c93686a45:63:18
    at http://localhost:9876/base/test/test_index.js?50c2d46de641c201d93f959a15507e2c93686a45:66:10
Chrome Headless 92.0.4515.107 (Linux x86_64) ERROR
  Uncaught TypeError: window.pbjs.processQueue is not a function
  at webpack:///test/test_index.js:8:26 <- test/test_index.js:74239:13
  
  TypeError: window.pbjs.processQueue is not a function
      at Object.<anonymous> (webpack:///test/test_index.js:8:26 <- test/test_index.js:74239:13)
      at __webpack_require__ (webpack:///webpack/bootstrap%206b65082d8b2d022eb522:19:1 <- test/test_index.js:20:30)
      at webpack:///webpack/bootstrap%206b65082d8b2d022eb522:62:1 <- test/test_index.js:63:18
      at test/test_index.js:66:10

Finished in 0.368 secs / 0 secs @ 15:55:21 GMT+0100 (British Summer Time)

SUMMARY:
✔ 0 tests completed
[15:55:21] 'testCoverage' errored after 7.12 s
[15:55:21] Error: Karma tests failed with exit code 1
    at /home/alex/src/adloox/Prebid.js/gulpfile.js:308:12
    at removeAllListeners (/home/alex/src/adloox/Prebid.js/node_modules/karma/lib/server.js:462:9)
    at /home/alex/src/adloox/Prebid.js/node_modules/karma/lib/server.js:469:9
    at Server.close (net.js:1608:9)
    at Object.onceWrapper (events.js:519:28)
    at Server.emit (events.js:412:35)
    at Server.emit (domain.js:532:15)
    at emitCloseNT (net.js:1661:8)
    at processTicksAndRejections (internal/process/task_queues.js:81:21)
[15:55:21] 'test-coverage' errored after 7.15 s
```

**Comment by jimdigriz** [comment]
@Rothalack @patmmccann it took about three times longer than it should have, but I have now the unit test covering this.

Please let me know what is missing.

**Comment by jimdigriz** [comment]
I have no idea why CircleCI is failing, it won't tell me without me providing it read/write access to all my public and private repositories which I am not prepared to do.

Running all the tests locally and through BrowserStack works fine for me though:
```
alex@aineko:~/src/adloox/Prebid.js$ gulp test --browserstack --nolint --nolintfix 
[10:29:10] Using gulpfile ~/src/adloox/Prebid.js/gulpfile.js
[10:29:10] Starting 'test'...
[10:29:10] Starting 'clean'...
[10:29:10] Finished 'clean' after 19 ms
[10:29:10] Starting 'lint'...
[10:29:10] Finished 'lint' after 1.14 ms
[snipped]
IE 11.0 (Windows 10): Executed 7333 of 7340 (skipped 11) SUCCESS (29.201 secs / 19.508 secs)
Edge 17.17134 (Windows 10): Executed 7333 of 7340 (skipped 11) SUCCESS (22.91 secs / 12.821 secs)
Edge 90.0.818.39 (Windows 10): Executed 7333 of 7340 (skipped 11) SUCCESS (15.25 secs / 8.443 secs)
Chrome 90.0.4430.72 (Windows 10): Executed 2927 of 7340 (skipped 8) SUCCESS (4.368 secs / 1.936 secs)
Chrome 79.0.3945.79 (Windows 10): Executed 7333 of 7340 (skipped 11) SUCCESS (17.432 secs / 9.55 secs)
Firefox 88.0 (Windows 10): Executed 2929 of 7340 (skipped 8) SUCCESS (4.509 secs / 2.12 secs)
Firefox 72.0 (Windows 10): Executed 7333 of 7340 (skipped 11) SUCCESS (16.349 secs / 10.548 secs)
Safari 14.1.1 (Mac OS 10.15.7): Executed 7333 of 7340 (skipped 11) SUCCESS (14.862 secs / 10.117 secs)
Safari 12.1.2 (Mac OS 10.14.6): Executed 7333 of 7340 (skipped 11) SUCCESS (18.798 secs / 12.813 secs)
TOTAL: 57187 SUCCESS

[10:36:23] Finished 'test' after 7.22 min
[10:36:23] Finished 'test' after 7.22 min

alex@aineko:~/src/adloox/Prebid.js$ gulp lint
[10:41:42] Using gulpfile ~/src/adloox/Prebid.js/gulpfile.js
[10:41:42] Starting 'lint'...
[10:42:03] Finished 'lint' after 21 s

alex@aineko:~/src/adloox/Prebid.js$ git log -n 2
commit ae7acd35807a1c3213ef2618c00e7fbdc67e4ef8 (HEAD -> fix/targeting-non-string, origin/fix/targeting-non-string, coremem/fix/targeting-non-string)
Author: Alexander Clouter <alex@coremem.com>
Date:   Sat Jun 19 14:08:09 2021 +0100

    targeting: allow non-string (eg. numeric) targeting segments
    
    Documentation[1] shows a numeric example which causes an exception as we
    try to call .split(',').
    
    [1] https://docs.prebid.org/dev-docs/add-rtd-submodule.html#gettargetingdata

commit 541fc70ada90703d628a0e520899900a6cecc73e (tag: 5.7.0)
Author: Matt Kendall <1870166+mkendall07@users.noreply.github.com>
Date:   Thu Jul 29 14:32:07 2021 -0400

    5.7.0 release
```

**Comment by ChrisHuie** [comment]
@jimdigriz sorry I am just seeing this thread. I sometimes lose my mentions in all the other github notifications. Re-running ci test

**Review by ChrisHuie — APPROVED** [review]

**Review by Rothalack — APPROVED** [review]

**Pull request merged**
